src/verilog.y

```diff
@@ -1419,13 +1416,17 @@ parameter_value_assignmentClass<pinp>:  // IEEE: [ parameter_value_assignment ]
 
 parameter_port_listE<nodep>:    // IEEE: parameter_port_list + empty == parameter_value_assignment
                 /* empty */                             { $$ = nullptr; }
-        |       '#' '(' ')'                             { $$ = nullptr; }
+        |       '#' '(' ')'                             { $$ = nullptr; GRAMMARP->m_modp->hasParameterList(true); }
```

Comment:
m_modp might be null on an error, for example a local param declared after a class (due to the m_modp bug I point out in another comment).  I'd suggest add a GRAMMARP-> helper function that gets a fileline and asserts that m_modp is non-null.

Likewise elsewhere.

```suggestion
        |       '#' '(' ')'
                        { $$ = nullptr; GRAMMARP->getModp($1)->hasParameterList(true); }
```

Reply:
I checked that the `parameter_port_listE` is only used by `class_declaration`, `module_declaration`, `interface_declaration` and `program_declaration`, both of them set m_modp properly. So there shouldn't be any problem.

Except that `udpFront` didn't set for it, but a UDP actually should not use the parameter list according to the LRM. So I remove the support for it by the way...